index.bs

```diff
@@ -5069,96 +5069,125 @@ No [=extended attributes=] are applicable to dictionaries.
+        <p>Let <var ignore>object</var> be a newly [=exception/created=] "{{SyntaxError}}"
+        {{DOMException}}.
+    </blockquote>
+
+    <p>To [=reject=] a promise with a new {{DOMException}} with [=error name=] "{{OperationError}}":
+
+    <blockquote>
+        [=Reject=] <var ignore>p</var> with an "{{OperationError}}" {{DOMException}}.
     </blockquote>
 </div>
 
+<div class=example id=example-exception-throwing-message-hint>
+    <p>An example of including additional information used to construct the [=exception/message=]
+    would be:
 
-<h4 id="idl-DOMException-error-names">Error names</h4>
+    <blockquote>
+        [=exception/Throw=] a "{{SyntaxError}}" {{DOMException}} indicating that the given value
+        had disallowed trailing spaces.
+    </blockquote>
 
-The <dfn id="dfn-error-names-table" export>error names table</dfn> below lists all the allowed error names
-for {{DOMException}}, a description, and legacy code values.
+    <p>Such additional context is most helpful to implementers when it is not immediately obvious
+    why the exception is being thrown, e.g., because there are many different steps in the algorithm
+    which throw a "{{SyntaxError}}" {{DOMException}}. In contrast, if your specification throws a
+    "{{NotAllowedError}}" {{DOMException}} immediately after checking if the user has provided
+    permission to use a given feature, it's fairly obvious what sort of [=exception/message=] the
+    implementation should construct, and so specifying it is not necessary.
+</div>
 
-<p class="warning">
-    The {{DOMException}} names marked as deprecated are kept for legacy purposes but their usage is discouraged.
-</p>
+The resulting behavior from creating and throwing an exception is language binding-specific.
+
+<p class=note>See [[#es-creating-throwing-exceptions]] for details on what creating and throwing an
+exception entails in the ECMAScript language binding.
+
+
+<h4 id="idl-DOMException-error-names">Base {{DOMException}} error names</h4>
+
+The <dfn id="dfn-error-names-table" export>error names table</dfn> below lists all the allowed error
+names for instances of the base {{DOMException}} interface, along with a description of what such
+error names mean, and legacy numeric error code values.
+
+<p class="note">Interfaces [=interface/inheriting=] from {{DOMException}}, in the manner described
+in [[#idl-DOMException-derived-interfaces]], will have their own names, not listed in this table.
 
-Note: If an error name is not listed here, please file a bug as indicated at the top of this specification and it will be addressed shortly. Thanks!
+When [=exception/creating=] or [=exception/throwing=] a {{DOMException}}, specifications must use
+one of these names. If a specification author believes none of these names are a good fit for their
+case, they must
+<a href="https://github.com/heycam/webidl/issues/new?title=DOMException%20name%20proposal">file an issue</a>
```

Comment:
```suggestion
<a href="https://github.com/whatwg/webidl/issues/new?title=DOMException%20name%20proposal">file an issue</a>
```

Reply:
Also fixing all the others in the document